### PR TITLE
tagged_nodes_by_author method(resolves  #1776)

### LINKED ISSUE #1776: Create a Node.tagged_nodes_by_author method
This has been marked as a good candidate for becoming a [first-timers-only issue](http://www.firsttimersonly.com) like [these](https://github.com/publiclab/plots2/labels/first-timers-only), meaning that it's simple, self-contained, and with some extra formatting and code pointers, could be a great entry point for a new contributor. 

If you're familiar enough with this code, please consider [reformatting or reposting it as a first-timers-only issue](https://publiclab.org/notes/warren/10-31-2016/create-a-welcoming-first-timers-only-issue-to-invite-new-software-contributors), and then ping @jywarren to get it labelled. Or, if this is not your first time, try to solve it yourself!

****

**Template draft:** (please leave additions in the comments to get this closer to a full first-timers-only issue)

Hi, this is a [first-timers-only issue](https://github.com/publiclab/plots2/projects/2). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

If you have contributed before, consider leaving this one for someone new, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

## Problem

We need a way to select all nodes which:

1. are authored or coauthored by a given user
2. are tagged with a given string like `tagname` (for example `replication:1234`)

This can be used, for example, to list activities completed by a given user (to help in completing #928).

## Solution

The format for calling it could be, perhaps:

`Node.tagged_nodes_by_author(tagname, user_id) # nodes by user with given tag`

This method could go in the Tag model, here:

https://github.com/publiclab/plots2/blob/master/app/models/tag.rb

## Steps to Fix

- [ ] claim this issue with a comment here, below, and ask any clarifying questions you need
- [ ] set up a repository locally following the README instructions, and make sure that all tests pass
- [ ] try to fix the issue following the steps above, but even before you're done, you can:
- [ ] commit your changes and start a pull request (see [contributing to Public Lab software](https://publiclab.org/wiki/contributing-to-public-lab-software)) but mark it as "in progress" if you have questions or if you haven't finished
- [ ] alert someone via the developers list (see below) to have your pull request merged. This may take a few extra steps depending on when you do it, but we'll help you out!

Please email the developers list (see https://publiclab.org/wiki/developers) or go to [the chatroom](http://publiclab.org/chat) if you have questions, and take a look at [our first-timers landing page for more information](https://github.com/publiclab/plots2/projects/2)!

### PR DESCRIPTION
**Pull request by SidharthBansal**
Added tagged_nodes_by_author method

Fixes #1776

### CODE REVIEW

**Comment by SidharthBansal** [comment]
Test is broken

**Comment by jywarren** [comment]
All right, i'll take a look!

File: app/models/tag.rb [hunk 1]
```diff
@@ -222,4 +222,11 @@ def self.trending(limit = 5 , start_date = DateTime.now - 1.month , end_date = D
        .order('count DESC')
        .limit(limit)
   end
+
+  #select nodes by tagname and user_id
+ def self.tagged_nodes_by_author(tagname, user_id)
+    Node.where('term_data.name = ?', tagname)
+       .includes(:node_tag, :tag)
```
Comment (jywarren) [inline app/models/tag.rb]: OK, in Rails 4, we also have to say `.references(:term_data)` if we're going to use that in a string-based `where()` filter. I'm getting used to this myself!

File: test/unit/node_test.rb [hunk 2]
```diff
@@ -297,4 +297,14 @@ class NodeTest < ActiveSupport::TestCase
     assert_equal note.likers.length, note.cached_likes
     assert_equal cached_likes-1 , note.cached_likes
   end
+
+  test "return nodes tagged by author and user_id" do
+    jeff = users(:jeff)
+    jeff_notes = nodes(:one)
+    jeff_notes.add_tag('replication:123',jeff)
+    tagged_note = Node.tagged_nodes_by_author('replication:123',2).first
```
Comment (jywarren) [inline test/unit/node_test.rb]: Here, this is my fault. I had originally said we should put it in `Node`, but then immediately on the next line of #1776 I suggested putting it in `Tag`. I think if you change `Node` to `Tag` this should be fixed. Thank you!!! And apologies!

File: app/models/tag.rb (hunk 1 above)
Comment (jywarren) [reply app/models/tag.rb]: I'm not 100% sure of this, so maybe make the other change first and we'll see how this does? If it still fails, then this change may also be needed. Also, if you could get all the `.` periods to align, it really does help keep the codebase tidy, and I'd appreciate it! Thanks!

**Comment by jywarren** [comment]
Looks great! Super job -- thanks!

**Pull request merged**